server/src/main/java/org/elasticsearch/action/synonyms/PutSynonymRuleAction.java

```diff
@@ -60,10 +61,11 @@ public Request(StreamInput in) throws IOException {
 
         public Request(String synonymsSetId, String synonymRuleId, BytesReference content, XContentType contentType) throws IOException {
             this.synonymsSetId = synonymsSetId;
-            this.synonymRule = PARSER.apply(
-                XContentHelper.createParser(XContentParserConfiguration.EMPTY, content, contentType),
-                synonymRuleId
-            );
+            try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, content, contentType)) {
+                this.synonymRule = PARSER.apply(parser, synonymRuleId);
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Failed to parse: " + content.utf8ToString(), e);
```

Comment:
Should the content we're returning in the log message be truncated here if it's too long?